steppy/base.py

```diff
@@ -162,22 +162,29 @@ class Step:
     """
 
     def __init__(self,
-                 name,
                  transformer,
-                 experiment_directory,
+                 name=None,
+                 experiment_directory=None,
                  input_data=None,
                  input_steps=None,
                  adapter=None,
-                 is_trainable=False,
-                 cache_output=False,
-                 persist_output=False,
+
+                 is_fittable=True,
+                 force_fitting=True,
+                 cache_output=True,
+                 persist_output=True,
+
                  load_persisted_output=False,
-                 force_fitting=False,
                  persist_upstream_pipeline_structure=False):
 
-        assert isinstance(name, str), 'Step name must be str, got {} instead.'.format(type(name))
-        assert isinstance(experiment_directory, str), 'Step {} error, experiment_directory must ' \
-                                                      'be str, got {} instead.'.format(name, type(experiment_directory))
+        name = self._validate_name(name, transformer)
```

Comment:
@kamil-kaczmarek I think the name of this method is a bit misleading. It creates new name if the name was not specified so maybe simply:
```python
if name is None:
   name = self._create_name()
```

Reply:
changed